test/jdk/javax/swing/regtesthelpers/Util.java

```diff
@@ -146,6 +147,26 @@ public static Component findSubComponent(Component parent, String className) {
         return null;
     }
 
+    /**
+     * Find a component based on predicate.
+     * Always run this method on the EDT thread
+     */
```

Comment:
The documentation could then explain the parameters, don't you think?

> Always run this method on the EDT thread

The utility method could ensure this automatically. This *public* method will be a wrapper around a *private* implementation, the implementation will run directly or on EDT using `invokeOnEDT` which already exists in the `Util` class.

Reply:
> If a wrapper is added we might need to check if it is already on EDT thread or not as below:

This is what I said: *“the implementation will run directly or on EDT.”* So if it's already on EDT, call directly; it it's not, use `invokeOnEDT`.

The utility method can then be used in all the contexts without thinking. (The implementation, if you choose this way, should rather be `findComponentImpl` rather than prefixed with an underscore.)

Likely, one still needs to call `findComponent` on EDT because any other operations on the component need to be on EDT. In this case, `findComponent` could throw an exception if it's called not on EDT.

These things will ensure the method can work correctly.

Reply:
Updated.